independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Methods.java

```diff
@@ -337,15 +337,24 @@ static class MethodKey {
         final List<DotName> params;
         final DotName returnType;
         final MethodInfo method; // this is intentionally ignored for equals/hashCode
+        private final int hashCode;
 
         public MethodKey(MethodInfo method) {
             this.method = Objects.requireNonNull(method, "Method must not be null");
             this.name = method.name();
             this.returnType = method.returnType().name();
-            this.params = new ArrayList<>();
-            for (Type i : method.parameterTypes()) {
-                params.add(i.name());
+            if (method.parameterTypes().size() == 0) {
```

Comment:
I think that you should use a local variable to hold the param types as well because `MethodInfo#parameterTypes()` creates a new immutable collection per each call.

Or maybe even something like:
```java
this.params = switch (method.parametersCount()) {
                case 0 -> List.of();
                case 1 -> List.of(method.parameterTypes().get(0).name());
                default -> {
                    List<DotName> ret = new ArrayList<>(method.parametersCount());
                    for (Type i : method.parameterTypes()) {
                        ret.add(i.name());
                    }
                    yield ret;
                }
            };
```

Reply:
Agree, I would certainly recommend using `method.parametersCount()` instead of `method.parameterTypes().size()`. Also @mkouba's `switch` looks better than the series of `if`s, I'd just recommend to add an option for _2_ parameters, as the immutable `List.of()` type has a variant for 1 and 2 elements.

Reply:
Done, thanks for the suggestions!